playbooks/roles/vault-ssh-signer/tasks/main.yml

```diff
@@ -0,0 +1,53 @@
+---
+- name: enable the {{ vault_ssh_signer_ca }} signer
+  shell: |
+    if vault secrets list | grep -q '^ssh-signer-{{ vault_ssh_signer_ca }}/' ; then
+      echo already set
+    else
+      vault secrets enable -path=ssh-signer-{{ vault_ssh_signer_ca }} ssh
+      echo signer created
+    fi
+  register: result
+  changed_when: "'signer created' in result.stdout"
+  environment: "{{ vault_env }}"
+
+- name: generate_signing_key for the {{ vault_ssh_signer_ca }} signer
+  shell: |
+    if vault read ssh-signer-{{ vault_ssh_signer_ca }}/config/ca ; then
+      echo already set
+    else
+      vault write ssh-signer-{{ vault_ssh_signer_ca}}/config/ca generate_signing_key=true
+      echo signing_key created
+    fi
+  register: result
+  changed_when: "'signing_key created' in result.stdout"
+  environment: "{{ vault_env }}"
+
+- name: copy the role definition
+  copy:
+    content: |
+      {
+        "allow_user_certificates": true,
+        "allowed_users": "*",
+        "default_extensions": [
+          {
+            "permit-pty": ""
+          }
+        ],
+        "key_type": "ca",
+        "default_user": "ubuntu",
+        "ttl": "720h0m0s"
+      }
```

Comment:
Possible for this whole JSON to be a variable? Might make it easier to reference / override where necessary later.

Reply:
Good idea, will do :+1:

Reply:
done